image/local.go

```diff
@@ -38,13 +38,13 @@ type local struct {
 	easyAddLayers    []string
 }
 
-func (f *Factory) NewLocal(repoName string) (Image, error) {
+func (f *Factory) NewLocal(repoName string) (*Local, error) {
```

Comment:
Looks like we forgot to change `NewEmptyLocal` as well? Also, should we remove the interface from this package?

I'd like to chat about the direction here before we merge.